SlicerNNUnet/SlicerNNUNetLib/InstallLogic.py

```diff
@@ -169,31 +169,43 @@ def _downgradeDynamicNetworkArchitecture(self) -> None:
         Workaround: fix incompatibility of dynamic_network_architectures==0.4 with totalsegmentator==2.0.5.
         Revert to the last working version: dynamic_network_architectures==0.2
         """
-        if parse(version("dynamic_network_architectures")) == parse("0.4"):
-            self._log(
-                f'dynamic_network_architectures package version is incompatible. Installing working version...')
-            self.pip_install("dynamic_network_architectures==0.2.0")
+        import sys
+        if sys.version_info < (3, 12):
+            # Python 3.9 (Slicer-5.8 and earlier)
+            if parse(version("dynamic_network_architectures")) == parse("0.4"):
+                self._log(
+                    f'dynamic_network_architectures package version is incompatible. Installing working version...')
+                self.pip_install("dynamic_network_architectures==0.2.0")
 
     def _installACVLUtils(self) -> None:
         """
         Workaround: fix incompatibility of acvl-utils 0.2.1 and 0.2.2 with nnunetv2.
         Revert to the last working version: acvl-utils==0.2.0
         """
-        # Recent versions of acvl_utils are broken:
-        # 0.2.1: https://github.com/MIC-DKFZ/acvl_utils/issues/2
-        # 0.2.2: https://github.com/MIC-DKFZ/acvl_utils/issues/4
-        # As a workaround, we install an older version manually. This workaround can be removed after acvl_utils is fixed.
+        import sys
+        if sys.version_info < (3, 12):
+            # Python 3.9 (Slicer-5.8 and earlier)
+            #   Recent versions of acvl_utils are broken:
+            #   0.2.1: https://github.com/MIC-DKFZ/acvl_utils/issues/2
+            #   0.2.2: https://github.com/MIC-DKFZ/acvl_utils/issues/4
+            #   As a workaround, we install an older version manually. This workaround can be removed after acvl_utils is fixed.
+            acvl_utils_version = "0.2"
+        else:
+            # Python >= 3.12 (Slicer-5.9 and later)
+            # acvl_utils 0.2.5 is usable
+            acvl_utils_version = "0.2.5"
```

Comment:
Should this version be fixed for python 3.12+? I would expect us to fallback to the nn-UNet library default required version if possible.

Reply:
Acvl-utils broke twice in about 2 years, which could justify pinning a specific version. But I agree that we could just let nnunet install it, as that would simplify things and there is a chance that not updating acvl-utils will cause problems.

Reply:
If we pin acvl utils, I think we should also pin the supported nnU-Net library to match. 
Unpinning everything should be the best way to have the latest nnU-Net version available for users with the limitation of having to manually fix things from time to time...

This problem is probably not limited to this package. I'm wondering if it would make sense to freeze versions along a Slicer release tag? Main version should point to the latest versions to have early regression warnings.